lib/beautiful/xresources.lua

```diff
@@ -83,8 +83,16 @@ function xresources.get_dpi(s)
         if awesome and awesome.xrdb_get_value then
             xresources.dpi = tonumber(awesome.xrdb_get_value("", "Xft.dpi"))
         end
+        -- Following Keith Packard's whitepaper on Xft,
+        -- https://keithp.com/~keithp/talks/xtc2001/paper/xft.html#sec-editing
+        -- the proper fallback for Xft.dpi is the vertical DPI reported by
+        -- the X server. This will generally be 96 on Xorg, unless the user
+        -- has configured it differently
         if not xresources.dpi then
-            xresources.dpi = 96
+            local root = require("root")
+            _, h = root.size()
+            _, hmm = root.size_mm()
+            xresources.dpi = round(h*25.4/hmm)
```

Comment:
I can define the constants `inches_to_millimeter` to explain the 25.4

As for testing, I'm actually running awesome with this code now, did I miss something?